environment/src/test/java/com/sequenceiq/environment/environment/flow/deletion/handler/StructuredEventCleanupHandlerTest.java

```diff
@@ -0,0 +1,86 @@
+package com.sequenceiq.environment.environment.flow.deletion.handler;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.sequenceiq.cloudbreak.util.TestConstants;
+import com.sequenceiq.environment.environment.dto.EnvironmentDeletionDto;
+import com.sequenceiq.environment.environment.dto.EnvironmentDto;
+import com.sequenceiq.environment.environment.flow.deletion.event.EnvDeleteEvent;
+import com.sequenceiq.environment.environment.service.cleanup.EnvironmentStructuredEventCleanupService;
+import com.sequenceiq.flow.reactor.api.event.EventSender;
+
+import reactor.bus.Event;
+import reactor.bus.Event.Headers;
+
+@ExtendWith(MockitoExtension.class)
+class StructuredEventCleanupHandlerTest {
```

Comment:
While not a frequent one, could you please add a test case for the `exceptionProcessor.handle()` scenario as well?